src/components/forms/ValidatedFormCheckbox.tsx

```diff
@@ -0,0 +1,60 @@
+import React from 'react';
+import omit from 'lodash/omit';
+import isString from 'lodash/isString';
+
+import { Form } from '@edx/paragon';
+
+import { setFormFieldAction } from './data/actions';
+import { useFormContext } from './FormContext';
+
+type InheritedParagonCheckboxProps = {
+  className?: string;
+  children: string;
+};
+
+export type ValidatedFormCheckboxProps = {
+  // Field id, required to map to field in FormContext
+  formId: string;
+  // Inline Instructions inside form field when blank
+  fieldInstructions?: string;
+  label?: string;
+  options?: string[][];
+  isInline?: boolean;
+} & InheritedParagonCheckboxProps;
+
+const ValidatedFormCheckbox = (props: ValidatedFormCheckboxProps) => {
+  const {
+    showErrors, formFields, errorMap, dispatch,
+  } = useFormContext();
+  const onChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    if (dispatch) {
+      dispatch(setFormFieldAction({ fieldId: props.formId, value: e.target.checked }));
+    }
+  };
+  const value = formFields?.[props?.formId];
+  const errors = errorMap?.[props.formId];
+  // Show error message if an error message was part of any detected errors
+  const showError = errors?.find?.(error => isString(error));
+  const formCheckboxProps = {
+    ...omit(props, ['formId']),
+    onChange,
+    isInvalid: showErrors && showError,
+    id: props.formId,
+    value: formFields && formFields[props.formId],
+  };
+
+  return (
+    <>
+      <Form.Label {...formCheckboxProps}>{formCheckboxProps.label}</Form.Label>
+      <Form.Checkbox {...formCheckboxProps} checked={value} onChange={onChange} />
+      {formCheckboxProps.fieldInstructions && (
+        <Form.Text>{formCheckboxProps.fieldInstructions}</Form.Text>
+      )}
+      {showErrors && showError && (
```

Comment:
Why is there showErrors twice?

Reply:
Ok reading this again, I see that its two different variables, but I think the {showError} below is confusing cause I would assume that's a boolean and not an error message

Reply:
That's a good point, went ahead and renamed the variable.